app/jobs/run_submissions_job.rb

```diff
@@ -0,0 +1,28 @@
+class RunSubmissionsJob < ApplicationJob
+  queue_as :submission
+
+  def perform(*submission_ids)
+    submissions = Submission.find(submission_ids)
+    submissions.each do |submission|
+      run_submission submission
+    end
+  end
+
+  def run_submission(submission)
+    tests = Test.where(problem_id: submission.problem_id)
+
+    tests.each do |test|
+      execution = Execution.find_or_initialize_by(submission_id: submission.id, test_id: test.id)
+      
+      if test.expected_output.nil?
```

Comment:
Yeah, I was also thinking that the library should be purely functional(not modify state). Didn't want to make too many changes in this PR. I'll make another PR and come back to this once it's fixed.